geode-core/src/test/java/org/apache/geode/internal/cache/partitioned/rebalance/PartitionedRegionLoadModelJUnitTest.java

```diff
@@ -483,30 +485,73 @@ public void testColocationEnforceLocalMaxMemory() throws UnknownHostException {
   }
 
   /**
-   * Test that each region individually honors it's enforce local max memory flag.
+   * Test that a region with enforceLocalMaxMemory disabled colocated with
+   * a region that memory full with enforceLocalMaxMemory enabled will prevent a bucket move.
    */
   @Test
-  public void testColocationIgnoreEnforceLocalMaxMemory() throws UnknownHostException {
+  public void testColocationOneNoneEvictionRegionReachLocalMaxMemoryLimit() throws UnknownHostException {
     PartitionedRegionLoadModel model = new PartitionedRegionLoadModel(bucketOperator, 1, 4,
         getAddressComparor(false), Collections.emptySet(), partitionedRegion);
     InternalDistributedMember member1 =
         new InternalDistributedMember(InetAddress.getByName("127.0.0.1"), 1);
     InternalDistributedMember member2 =
         new InternalDistributedMember(InetAddress.getByName("127.0.0.1"), 2);
 
-    // Create some buckets with low redundancy on member 1
     PartitionMemberInfoImpl details1 =
-        buildDetails(member1, new long[] {1, 1, 1, 1}, new long[] {1, 1, 1, 1});
+        buildDetails(member1, 1, 8 * MB, new long[] {1 * MB, 1 * MB, 1 * MB, 1 * MB},
+            new long[] {1, 1, 1, 1});
     PartitionMemberInfoImpl details2 =
-        buildDetails(member2, new long[] {0, 0, 0, 0}, new long[] {0, 0, 0, 0});
+        buildDetails(member2, 1, 8 * MB, new long[] {0, 0, 0, 0}, new long[] {0, 0, 0, 0});
+    model.addRegion("a", Arrays.asList(details1, details2), new FakeOfflineDetails(), false);
+
+
+    PartitionMemberInfoImpl bDetails1 =
+        buildDetails(member1, 1, 2 * MB, new long[] {1 * MB, 1 * MB, 1 * MB, 1 * MB},
+            new long[] {1, 1, 1, 1});
+    PartitionMemberInfoImpl bDetails2 =
+        buildDetails(member2, 1, 2 * MB, new long[] {0, 0, 0, 0}, new long[] {0, 0, 0, 0});
+    model.addRegion("b", Arrays.asList(bDetails1, bDetails2), new FakeOfflineDetails(), true);
+
+
+    assertThat(doMoves(new CompositeDirector(true, true, false, true), model)).isEqualTo(4);
+
+    Set<Create> expectedCreates = new HashSet<>();
+    expectedCreates.add(new Create(member2, 0));
+    expectedCreates.add(new Create(member2, 1));
+    assertThat(new HashSet<>(bucketOperator.creates)).isEqualTo(expectedCreates);
+
+    Set<Move> expectedMoves = new HashSet<>();
+    expectedMoves.add(new Move(member1, member2));
+    expectedMoves.add(new Move(member1, member2));
+    assertThat(new HashSet<>(bucketOperator.primaryMoves)).isEqualTo(expectedMoves);
+  }
+
+  /**
+   * Test that a region that memory full with enforceLocalMaxMemory disabled will not prevent a
+   * bucket move.
+   */
+  @Test
+  public void testColocationOneEvictionRegionReachLocalMaxMemoryLimit() throws UnknownHostException {
```

Comment:
I would rename the test case to: `testColocationOneEvictionRegionReachesLocalMaxMemoryLimit()`